complex/src/lib.rs

```diff
@@ -150,6 +152,48 @@ impl<T: Clone + Float> Complex<T> {
         let (r, theta) = self.to_polar();
         Complex::from_polar(&(r.sqrt()), &(theta/two))
     }
+    
+    /// Raises `self` to a floating point power.
+    #[inline]
+    pub fn powf(&self, exp: T) -> Complex<T> {
+        let (r, theta) = self.to_polar();
+        Complex::from_polar(&r.powf(exp), &(theta*exp))
+    }
+    
+    /// Returns the logarithm of `self` with respect to an arbitrary base.
+    #[inline]
+    pub fn log(&self, base: T) -> Complex<T> {
+        let (r, theta) = self.to_polar();
+        Complex::new(r.log(base), theta * T::one() / base.ln())
```

Comment:
> You don't need the one here, just `theta / base.ln()`.

There are times I feel stupid and sometimes this feeling is well justified. Thanks for the note I will update it soon!